src/DRAMPower/DRAMPower/memspec/MemSpecLPDDR5.cpp

```diff
@@ -95,6 +98,41 @@ uint64_t MemSpecLPDDR5::timeToCompletion(DRAMPower::CmdType type)
     return offset;
 }
 
+void MemSpecLPDDR5::parseImpedanceSpec(nlohmann::json &memspec) {
+    if (!memspec.contains("memimpedancespec")) {
+        // Leaving it to default-initialize to 0 would break static power (div by 0)
+        memImpedanceSpec = {1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0};
+        return;
+    }
 
+    memImpedanceSpec.C_total_cb =
+        parseUdouble(memspec["memimpedancespec"]["C_total_cb"], "C_total_cb");
+    memImpedanceSpec.C_total_ck =
+        parseUdouble(memspec["memimpedancespec"]["C_total_ck"], "C_total_ck");
+    memImpedanceSpec.C_total_wck =
+        parseUdouble(memspec["memimpedancespec"]["C_total_wck"], "C_total_wck");
+    memImpedanceSpec.C_total_dqs =
+        parseUdouble(memspec["memimpedancespec"]["C_total_dqs"], "C_total_dqs");
+    memImpedanceSpec.C_total_rb =
+        parseUdouble(memspec["memimpedancespec"]["C_total_rb"], "C_total_rb");
+    memImpedanceSpec.C_total_wb =
+        parseUdouble(memspec["memimpedancespec"]["C_total_wb"], "C_total_wb");
+
+    memImpedanceSpec.R_eq_cb = parseUdouble(memspec["memimpedancespec"]["R_eq_cb"], "R_eq_cb");
+    memImpedanceSpec.R_eq_ck = parseUdouble(memspec["memimpedancespec"]["R_eq_ck"], "R_eq_ck");
+    memImpedanceSpec.R_eq_wck = parseUdouble(memspec["memimpedancespec"]["R_eq_wck"], "R_eq_wck");
+    memImpedanceSpec.R_eq_dqs = parseUdouble(memspec["memimpedancespec"]["R_eq_dqs"], "R_eq_dqs");
+    memImpedanceSpec.R_eq_rb = parseUdouble(memspec["memimpedancespec"]["R_eq_rb"], "R_eq_rb");
+    memImpedanceSpec.R_eq_wb = parseUdouble(memspec["memimpedancespec"]["R_eq_wb"], "R_eq_wb");
+}
 
+void MemSpecLPDDR5::parseDataRateSpec(nlohmann::json &memspec) {
```

Comment:
Is this spec required? The rate between CK and data transmission rate (2*WCK) is specified as dataRate in the memarchitecturespec. And both DQ as well as DQS are running at those rates. Or is there a mode where they can have different rates?

Reply:
I don`t think so. I removed the spec, since I copied it over from LPDDR4